src/components/portal/portal.tsx

```diff
@@ -159,13 +159,13 @@ export class Portal {
         }
 
         if ('ResizeObserver' in window) {
-            const observer = new ResizeObserver(() => {
+            this.observer = new ResizeObserver(() => {
                 if (this.popperInstance) {
                     this.styleContainer();
                     this.popperInstance.update();
                 }
             });
-            observer.observe(this.container);
+            this.observer.observe(this.container);
```

Comment:
Without this change, the cleanup on line 125 doesn't really do much… 🙈

<img width="520" alt="image" src="https://github.com/user-attachments/assets/75c2c1a1-805f-473e-9e8a-238bf412d81f">